### PR TITLE
[ML][HLRC] Add data frame analytics regression analysis

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**

### CODE REVIEW

**Review by benwtrent — COMMENTED** [review]
Just a couple of things.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientTests.java [hunk 1]
```diff
@@ -711,8 +710,9 @@ public void testProvidedNamedXContents() {
         assertTrue(names.contains(ShrinkAction.NAME));
         assertTrue(names.contains(FreezeAction.NAME));
         assertTrue(names.contains(SetPriorityAction.NAME));
-        assertEquals(Integer.valueOf(1), categories.get(DataFrameAnalysis.class));
+        assertEquals(Integer.valueOf(2), categories.get(DataFrameAnalysis.class));
         assertTrue(names.contains(OutlierDetection.NAME.getPreferredName()));
+        assertTrue(names.contains(org.elasticsearch.client.ml.dataframe.Regression.NAME.getPreferredName()));
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientTests.java]: Does it have to be fully-qualified?

**Review by przemekwitek — APPROVED** [review]
LGTM

**Review by benwtrent — APPROVED** [review]

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientTests.java (hunk 1 above)
Comment (dimitris-athanasiou) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/RestHighLevelClientTests.java]: Yes, there is also the `Regression` object from the evaluation API.

**Comment by dimitris-athanasiou** [comment]
@elasticmachine update branch

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/1

**Pull request merged**